openapi.yml

```diff
@@ -32,65 +51,74 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/OrderFull"
-  /api/v1/nonce:
+  /api/v1/gas-tip/{sellToken}:
     get:
-      summary: The nonce to pick for a new order.
-      description: The best nonce to pick depends on how much time there is left in the current settlement according to the operator. This time is token-pair specific which is why the sell and buy tokens must be supplied.
-      parameters:
-        - name: sellToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
-        - name: buyToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
+      summary: The tip that is used to pay solvers for their service in the sellToken of an order.
+      description: The tip is calculated by the order service and depends on the current gas price and the expected gas costs to settle the trade
       responses:
         200:
-          description: the nonce
+          description: the tip
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Nonce"
+                $ref: "#/components/schemas/Tip"
+        400:
+          description: sellToken not existent
 components:
   schemas:
     Address:
       description: Ethereum 40 byte address encoded as a hex without `0x` prefix.
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
-      type: integer
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
+      type: object
+      properties:
+        expirationDate:
+          description: Expiration date of the offered tip. Encoded as ISO 8601 UTC.
+          type: string
+          example: "2020-10-22T07:57:18Z"
```

Comment:
How is this intended to work with the settlement contract? The tip is signed with the order parameters and currently doesn't have a separate expiration date from the actual order.

Reply:
No, it is not meant to have any impact on smart contracts.

Anxo asked for it with the following intend:
>Also, I think the operator should return also a date together with the tip (expirationDate). So the clients know until when the operator would allow receiving orders for that sell token with that minimal tip.

If I understand @anxolin correctly, then they would need it in the following scenario: The user asks to sends an order and the FE asks for the tip. Then we would have one "confirm module" and the signing process. If the user then takes too long, longer than the specified expiration time to confirm the order, the order service will not accept the order. The front-end can prevent this case if it is aware of an expiration time.

Alternatively, we could also assume that users don't need longer than 3 minutes for the confirmation module and the signing process and the order service just allows the tip that it has proposed within the last 3 minutes for this particular order.  Then, in the same scenario: "the user takes longer than 3 minutes", the front-end could also check the tip again right at the moment the end of the signing process and warn the user, if the new required tip is higher than the old one. THen the FE would have to restart with a new confirm module and a higher tip.

Reply:
Oh, its not part of the `Order` type :see_no_evil:.

Makes perfect sense! Sorry!

Reply:
Yes, we won't be querying the tip all the time. Only when you first select the token.
then depending on your commitment (`expireDate`) we won't be asking again in a while. 

`IMPORTANT`: for when you implement this, if u have some kind of dynamic calculation, please keep in mind that we should do roundings in the backend for the tip, so we keep it somewhat stable. This is because, if it changes while the user is confirming the order, it will ask him confirmation if he accepts the new price. For this reason, I think we shouldn't be changing on tiny price fluctuations. Rounding up to 1$?